atlasdb-cassandra-integration-tests/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraStrangeBehaviourTest.java

```diff
@@ -121,4 +130,21 @@ private void createTablesIgnoringException() {
                 TABLES.stream().collect(Collectors.toMap(x -> x, no -> AtlasDbConstants.GENERIC_TABLE_METADATA))))
                 .isInstanceOf(IllegalStateException.class);
     }
+
+    private void createTableWithMetadataInLegacyCell(TableReference tableRef) {
+        kvs.createTable(tableRef, AtlasDbConstants.GENERIC_TABLE_METADATA);
+
+        Cell metadataCell = CassandraKeyValueServices.getMetadataCell(tableRef);
+        kvs.deleteAllTimestamps(AtlasDbConstants.DEFAULT_METADATA_TABLE,
+                ImmutableMap.of(metadataCell, Long.MAX_VALUE),
+                true);
+
+        Cell legacyMetadataCell = CassandraKeyValueServices.getOldMetadataCell(tableRef);
+        kvs.put(AtlasDbConstants.DEFAULT_METADATA_TABLE,
+                ImmutableMap.of(legacyMetadataCell, AtlasDbConstants.GENERIC_TABLE_METADATA),
+                System.currentTimeMillis());
+
+        assertThat(kvs.getMetadataForTable(LOWER_UPPER)).isNotEmpty();
+        assertThat(kvs.getMetadataForTables()).isNotEmpty();
+    }
```

Comment:
There is one sad thing we can do

```java
    @Test
    public void legacyCellWithDifferentCapitalisationBreaksTheWorld() {
        Cell legacyMetadataCell = CassandraKeyValueServices.getOldMetadataCell(LOWER_UPPER);
        kvs.put(AtlasDbConstants.DEFAULT_METADATA_TABLE,
                ImmutableMap.of(legacyMetadataCell, AtlasDbConstants.GENERIC_TABLE_METADATA),
                System.currentTimeMillis());
        kvs.createTable(LOWER_LOWER, AtlasDbConstants.GENERIC_TABLE_METADATA);
        kvs.dropTable(LOWER_LOWER);

        assertThat(kvs.getMetadataForTable(LOWER_LOWER)).isNotEmpty(); // wat
    }
```

but the methods I'm aware of to fix this require considering O(2^n) elements, so there's probably nothing much reasonable we can do

Reply:
Arguably, as long as there are entries there we should really clean up. I can follow up later